trin-core/src/jsonrpc/types.rs

```diff
@@ -274,8 +275,11 @@ impl TryFrom<[&Value; 2]> for OfferParams {
                     .collect();
 
                 if let Ok(content_keys) = content_keys {
-                    let content_keys: Result<Vec<Vec<u8>>, _> =
-                        content_keys.iter().map(hex::decode).collect();
+                    let content_keys: Result<Vec<Vec<u8>>, _> = content_keys
+                        .iter()
+                        .map(|s| s.as_str())
+                        .map(hex_decode)
```

Comment:
I guess we can get rid of the second `map` with something like `.map(|s| hex_decode(s.into())`.

Reply:
`into()` complained, but `as_str()` seems to work:
```rs
error[E0277]: the trait bound `&str: std::convert::From<&std::string::String>` is not satisfied
   --> trin-core/src/jsonrpc/types.rs:280:47
    |
280 |                         .map(|s| hex_decode(s.into()))
    |                                               ^^^^ the trait `std::convert::From<&std::string::String>` is not implemented for `&str`
    |
    = help: the following implementations were found:
              <&'a str as std::convert::From<&'a ascii::ascii_str::AsciiStr>>
              <&'a str as std::convert::From<webpki::name::dns_name::DnsNameRef<'a>>>
              <&'t str as std::convert::From<regex::re_unicode::Match<'t>>>
    = note: required because of the requirements on the impl of `Into<&str>` for `&std::string::String`
```